app/src/editor/canvas/components/Ports.jsx

```diff
@@ -503,9 +511,16 @@ export default class Ports extends React.Component {
         }
 
         // dynamically size port controls based on largest value
-        const portSize = Math.min(module.params.reduce((size, { value, defaultValue }) => (
-            Math.max(size, String(value || defaultValue).length)
-        ), Math.max(4, this.state.minPortSize)), 40)
+        const portSize = Math.min(module.params.reduce((size, { value, defaultValue, possibleValues }) => {
+            // For select, figure out width based the on the label instead of value
+            let label = value || defaultValue
+
+            if (possibleValues) {
+                label = (possibleValues.find((option) => option.value === value) || {}).name || value
+            }
+
+            return Math.max(size, String(label).length)
+        }, Math.max(4, this.state.minPortSize)), 40)
```

Comment:
Can we get the label-based port-sizing logic into a method rather than duplicating?

Reply:
Yep, add a `getPortDisplayValueSize` method with 2 character padding for selects so the arrows don't hide the value.